timeslots/handler.go

```diff
@@ -35,18 +46,35 @@ func BookTimeslot(w http.ResponseWriter, r *http.Request) {
 	var data Timeslot
 	json.NewDecoder(r.Body).Decode(&data)
 
+	// TODO: data.Time is empty here
 	ts, err := repo.getTimeslot(data)
 
 	if err != nil {
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
-	if ts.Booked {
+	var foundTimeslot Timeslot
+
+	for i := 0; i < len(ts); i++ {
+		var time TimePeriod
+		marshalError := json.Unmarshal(ts[i].Time, &time)
```

Comment:
consider to move TimePeriod marshaling/unmarshaling into database layer (repository)